_posts/2023-02-02-easy-cpu-gpu.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title: Easy CPU/GPU Arrays and Dataframes
+author: Rick Zamora (NVIDIA), Ben Zaitlen (NVIDIA)
+tags: []
+theme: twitter
+canonical_url: https://medium.com/rapids-ai/easy-cpu-gpu-arrays-and-dataframes-run-your-dask-code-where-youd-like-e349d92351d
+---
+
+_Original version of this post appears on https://medium.com/rapids-ai/easy-cpu-gpu-arrays-and-dataframes-run-your-dask-code-where-youd-like-e349d92351d_
+
+It's now easy to switch between CPU (NumPy / Pandas) and GPU (CuPy / cuDF) in Dask.
+As of Dask 2022.10.0, users can optionally select the backend engine for input IO and data creation. In the short-term, the goal of the backend-configuration system is to enable Dask users to write code that will run on both CPU and GPU systems.
+
+The preferred backend can be configured using the `array.backend` and `dataframe.backend` options with the standard Dask configuration system:
+
+```python
+dask.config.set({"array.backend": "cupy"})
+dask.config.set({"dataframe.backend": "cudf"})
+```
+
+## Dispatching for Array Creation
+
+To see how users can easily switch between NumPy and CuPy, let's start by creating an array of ones:
+
+```python
+>>> with dask.config.set({"array.backend": "cupy"}):
+...     darr = da.ones(10, chunks=(5,))  # Get cupy-backed collection
+...
+>>> darr
+dask.array<ones_like, shape=(10,), dtype=float64, chunksize=(5,), chunktype=cupy.ndarray>
+```
+
+The `chunktype` informs us that the array is constructed with `cupy.ndarray`
+objects instead of `numpy.ndarray` objects.
+
+We've also improved the user experience for random array creation. Previously, if a user wanted to create a CuPy-backed Dask array, they were required to define an explicit `RandomState` object in Dask using CuPy. For example, the following code worked prior to Dask 2022.10.0, but seems rather verbose:
+
+```python
+>>> import cupy
+>>> import dask.array as da
+>>> rs = da.random.RandomState(RandomState=cupy.random.RandomState)
+>>>
+>>> darr = rs.randint(0, 3, size=(10, 20), chunks=(2, 5))
+>>> darr
+dask.array<randint, shape=(10, 20), dtype=int64, chunksize=(2, 5), chunktype=cupy.ndarray>
+```
+
+Now, we can leverage the `array.backend` configuration to create a CuPy-backed dask array for random data:
+
+```python
+>>> with dask.config.set({"array.backend": "cupy"}):
+...     darr = da.random.randint(0, 3, size=(10, 20), chunks=(2, 5))  # Get cupy-backed collection
+...
```

Comment:
Noting the indention level looks normal when this markdown file is rendered on GitHub, but looks off on the NVIDIA post 

![Screenshot 2023-02-01 at 2 43 59 PM](https://user-images.githubusercontent.com/11656932/216159073-0554d5d3-b753-4f96-ab86-8dd85d9e1056.png)

Reply:
Thanks for pointing this out @jrbourbeau .  It should be fixed now